meson_options.txt

```diff
@@ -6,3 +6,4 @@ option('api-docs', type : 'feature', description : 'enable to generate API docum
 option('gtkdoc', type : 'boolean', description : 'enable to generate gtk-doc compatible headers for the API')
 option('python-command', type: 'string', description: 'the default Python command')
 option('socket', type: 'boolean', description: 'enable if you want to detect a running instance')
+option('mac-integration', type: 'boolean', value: 'false', description: 'enable for improved macOS integration using the gtk-mac-integration library')
```

Comment:
Probably in-line with autotools but I wonder why this is off by default? Couldn't this default to on and maybe dynamically detect presence of the dependency (or simply require it unless mac-integration is explicitly disabled)?

Reply:
Or maybe the default value is set by the existence of the dependency, so it can be forcibly overridden if detection fails.

Reply:
> Probably in-line with autotools but I wonder why this is off by default? Couldn't this default to on and maybe dynamically detect presence of the dependency (or simply require it unless mac-integration is explicitly disabled)?

At least for testing, I'd like to be able to manually override the auto-detected presence of the library. With autotools I just gave up ;-). I tried the following with meson:

```meson
mac_integration_dep = ['gtk-mac-integration', '3.0.1']
mac_integration = dependency(mac_integration_dep[0], version: '>= ' + mac_integration_dep[1],
	required: get_option('mac-integration').enabled())
deps_for_pc += ' ' + mac_integration_dep[0] + ' >= ' + mac_integration_dep[1]
```

assuming (based on the documentation) that `get_option('mac-integration').enabled()` would return `true` if explicitly enabled on the command-line and `false` if disabled or not specified. But I get
```
meson.build:27:0: ERROR: Unknown method "enabled" in object <[BooleanHolder] holds [bool]: True> of type BooleanHolder.
```
What am I doing wrong?

In any case, I don't think auto-enabling the library is that important. The library is probably not used by many applications and not normally installed and in fact, if Geany started using GtkApplication and GMenuModel, we could eliminate this dependency as its functionality is present in GTK now.

Reply:
`get_option('mac-integration')` already returns a boolean. But actually you'd use a [feature](https://mesonbuild.com/Reference-manual_returned_feature.html).

Features are an easier way to implement optional dependencies. They can be disabled on the command line (thereby skipping related dependencies) and you can pass them directly as required.

`mac_integration = dependency(mac_integration_dep[0], version: '>= ' + mac_integration_dep[1],	required: get_option('mac-integration'))`

Reply:
Thanks, fixed. Seems to work as advertised. Does the fixup look OK to you?

Reply:
yea